### PR TITLE
Clean up conversation view logic

### PR DESCRIPTION
**Pull request by jerith**

### CODE REVIEW

File: go/vumitools/conversation/utils.py [hunk 1]
```diff
@@ -39,18 +39,23 @@ def get_tagpool_metadata(self, key, default=None):
 
     @Manager.calls_manager
     def end_conversation(self):
-        # TODO: `stop' and `archive' should be different operations.
-        self.c.set_status_stopping()
+        # TODO: Remove this once the old UI is gone.
+        # Pretend we're archived, stop_conversation() saves for us.
```
Comment (hodgestar) [inline go/vumitools/conversation/utils.py:43]: Not quite sure what this comment means? Would `end_conversation` not eventually become `archive_conversation`?

**Comment by peterp** [comment]
:+1:

File: go/vumitools/conversation/utils.py [hunk 2]
```diff
@@ -39,18 +39,23 @@ def get_tagpool_metadata(self, key, default=None):
...
         self.c.set_status_finished()
-        yield self.c.save()
+        # Stop as normal
+        yield self.stop_conversation()
```
Comment (hodgestar) [inline go/vumitools/conversation/utils.py:46]: Won't calling `stop_conversation` here set the status to the wrong thing?

File: go/conversation/views.py [hunk 3]
```diff
@@ -86,14 +88,37 @@ def conversation_action(request, conversation_key, action_name):
...
+    conversation_type = form.cleaned_data['type']
+    conv = request.user_api.new_conversation(
+        conversation_type, name=form.cleaned_data['name'],
+        description=u'', config={})
+    messages.info(request, 'Conversation created successfully.')
+
+    view_def = get_conversation_view_definition(
+        conv.conversation_type, conv)
+
+    # TODO: Better workflow here?
```
Comment (hodgestar) [inline go/conversation/views.py:113]: Could you elaborate on this comment a bit?

File: go/conversation/view_definition.py [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+class StartConversationView(ConversationApiView):
+    view_name = 'start'
+    path_suffix = 'start/'
+
+    def post(self, request, conversation):
+        # raise NotImplementedError("TODO: Finish implementing this.")
+        profile = request.user.get_profile()
+        account = profile.get_user_account()
+        if account.confirm_start_conversation:
+            return self._start_via_confirmation(request, account, conversation)
```
Comment (hodgestar) [inline go/conversation/view_definition.py:61]: This confirmation stuff will have to get hooked up to the message sending action. I created #377 to remind us.

File: go/conversation/view_definition.py [hunk 5]
```diff
@@ -0,0 +1,435 @@
...
+class ConversationApiView(ConversationViewMixin, View):
+    pass
+
+
+class StartConversationView(ConversationApiView):
+    view_name = 'start'
+    path_suffix = 'start/'
+
+    def post(self, request, conversation):
+        # raise NotImplementedError("TODO: Finish implementing this.")
```
Comment (hodgestar) [inline go/conversation/view_definition.py]: I guess this line can just be deleted?

File: go/vumitools/conversation/utils.py (hunk 1 above)
Comment (jerith) [inline go/vumitools/conversation/utils.py:43]: `end_conversation` is part of the old conversation lifecycle. We can get rid of a lot of stuff like that when we switch over to the new UI completely. Until then, I'd like to minimise the differences between develop and develop-frontend, especially in non-UI code.

File: go/vumitools/conversation/utils.py (hunk 2 above)
Comment (jerith) [inline go/vumitools/conversation/utils.py:46]: `stop_conversation()` only sets `status`, `set_status_finished()` only sets `archive_status`.

File: go/conversation/views.py (hunk 3 above)
Comment (jerith) [inline go/conversation/views.py:113]: I think we need a better way of determining which screen to go to next, but until we have more bits hooked up in the right places I'm not sure what the requirements are.

**Comment by hodgestar** [comment]
I'm :+1:. Still a lot of cleaning up to go but we can't do it all in one PR.

File: go/conversation/view_definition.py (hunk 5 above)
Comment (jerith) [inline go/conversation/view_definition.py]: Yes, thanks.

**Pull request merged**